cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -199,21 +197,18 @@ auto make_chars_and_offsets(StringsIterator begin, StringsIterator end,
  *
  * @tparam Element The fixed-width element type
  *---------------------------------------------------------------------------**/
-template <typename Element>
+template <typename ElementTo, typename ElementFrom = ElementTo>
```

Comment:
Does `ElementFrom` really need to be a part of the class's type?

Can't the init list ctor be a template? It would be ideal to avoid having to specify the explicit type param in `fixed_width_column_wrapper`. 

https://wandbox.org/permlink/oEi9LrZe1sXAERoq

Reply:
Let me give it another shot. I also got a simplified example working in godbolt when I initially attempted, but the corresponding change did not succeed in the codebase. I will report back.

Reply:
@jrhemstad A couple of the issues I am working on right now (these are the easier ones):

https://wandbox.org/permlink/iHjJQmAIQDapHAW1

Reply:
Solution to issue 2:
```cpp
    // Issue 2
    using f = float;
    column_wrapper<float>  a{f(0.5), f(1.0), f(1.5), NAN};
    column_wrapper<double> b{0.5, 1.0, 1.5, double(NAN)};
```

Reply:
I think the fundamental problem is that currently, the `fixed_width_column_wrapper` constructors that take `initializer_list`s (and the current variant) don't use deduction. `Element` (from before) and now `ElementTo` specify the type. It means that:
```cpp
fixed_width_column_wrapper<TypeParam> col{1, 2, 3, 4.0, 5.0. 6.0};
```
will compile now and fails with the proposed change. Now, I have gone and fixed all the obvious ones that are similar to above - but there are some trickier ones that I am still trying to resolve (things to do with `timestamp`, etc). 

Will update later with progress.

Reply:
Issue 1 is just a result of the fact that I used `std::vector` in my example. If you replace with `thrust::host_vector` I'd expect it to work correct. 

Issue 2 I've seen before. I can't recall exactly right now, but there's some trickiness in the type deduction that's leading to ambiguity. I know I ran into this before with the column_wrapper. I'll have to look back and see if I can't find what I figured out before.

Reply:
Okay, so for issue 2, the problem is that the type `U` for the `std::initializer_list<U>` is now deduced (as you said), so you have to make sure everything in the list has the same type (since conversions aren't considered during type deduction). 

You're getting bit by this because `1.0` is type `double` and `NAN` is type `float`. Since those have different types, the compiler is complaining about deducing conflicting types: https://wandbox.org/permlink/Ji4XYOL6ACPZ7xTu 

So there's a few ways to solve that other than your proposed solution, e.g., you could also do:
```
     column_wrapper<float>  a{{0.5, 1.0, 1.5, std::numeric_limits<double>::quiet_NaN()}};
     column_wrapper<double> b{0.5, 1.0, 1.5, std::numeric_limits<double>::quiet_NaN()};
```

I imagine it's uncommon to try and specify objects of different types in the init list for a column wrapper. Are you finding this to be a common problem?

Reply:
Yes! I have it working. So it turns out the FINAL issue was the wonderful `{ }` vs `( )` construction. Now that the initializer list isn't told the type and has to deduce, if can't resolve the correct construction call with braces `{ }`, so every instance that uses them has to be replaced with this `( )`. It took me a while to figure out because the way that the problem manifested itself via compiler error message was super misleading.

Anyway, will push code tomorrow after cleaning it up.

Reply:
I've been bitten by that brace init list issue several times with `column_wrapper`. The compiler error really is super unhelpful.

Reply:
Changes have been pushed :)